presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -654,7 +697,8 @@ public void dropColumn(String databaseName, String tableName, String columnName)
                         .withDatabaseName(databaseName)
                         .withTableName(tableName)
                         .withExpression(expression)
-                        .withNextToken(nextToken));
+                        .withNextToken(nextToken)
+                        .withSegment(segment));
```

Comment:
it's hierarchical: next page (token) within a segment, so move `withSegment` before `withNextToken`

Reply:
OK